vector/src/main/java/im/vector/app/features/home/room/list/home/layout/HomeLayoutSettingBottomDialogFragment.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.home.room.list.home.layout
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import dagger.hilt.android.AndroidEntryPoint
+import im.vector.app.R
+import im.vector.app.core.platform.VectorBaseBottomSheetDialogFragment
+import im.vector.app.databinding.BottomSheetHomeLayoutSettingsBinding
+import im.vector.app.features.home.room.list.home.HomeLayoutPreferences
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class HomeLayoutSettingBottomDialogFragment : VectorBaseBottomSheetDialogFragment<BottomSheetHomeLayoutSettingsBinding>() {
+
+    @Inject lateinit var preferences: HomeLayoutPreferences
```

Comment:
This should be moved to a `ViewModel`, and the UI should be updated in the `invalidate` method.

Reply:
Not sure we'll benefit from VM here. This dialog is too small and too specific to make it's part reusable and too simple to require separation of concerns. Having VM here will just increase amount of boilerplate and make it less easy to understand. We also have plenty of other dialogs without VMs and I did think it's because of reasons above. Did we change approach and now require to follow same pattern no matter of what?

Reply:
I agree it's a bit more boilerplate code, but the idea is to have the logic in the ViewModel also to unit test it. Views (Fragment, etc.) should only be responsible of displaying things.
I guess for this particular part it can stay like that. Maybe better to spend effort to use DataStore.